modules/light-clients/08-wasm/types/vm.go

```diff
@@ -139,19 +157,72 @@ func wasmSudo[T ContractResult](ctx sdk.Context, clientStore storetypes.KVStore,
 	if err := json.Unmarshal(resp.Data, &result); err != nil {
 		return result, errorsmod.Wrap(ErrWasmInvalidResponseData, err.Error())
 	}
+
+	codeHash := cs.CodeHash
+	newClientState, err := validatePostExecutionClientState(clientStore, cdc)
+	if err != nil {
+		return result, err
+	}
+
+	// codehash should only be able to be modified during migration.
+	if !bytes.Equal(codeHash, newClientState.CodeHash) {
+		return result, errorsmod.Wrapf(ErrWasmInvalidContractModification, "expected code hash %s, got %s", hex.EncodeToString(codeHash), hex.EncodeToString(newClientState.CodeHash))
+	}
```

Comment:
@DimitrisJim suggested to do this check also in `wasmMigrate` and then it could be included in `validatePostExecutionClientState`, just to make sure the the contract is not doing something we don't expect, even though we set the code hash in 08-wasm after calling `wasmMigrate`.